src/System Application/App/AI/src/Azure AI Document Intelligence/AzureDIImpl.Codeunit.al

```diff
@@ -0,0 +1,156 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+namespace System.Azure.DI;
+using System.AI;
+using System.Globalization;
+using System.Privacy;
+using System.Telemetry;
+using System;
+
+/// <summary>
+/// Azure Document Intelligence implementation.
+/// </summary>
+codeunit 7779 "Azure DI Impl."
+{
+    Access = Internal;
+    InherentEntitlements = X;
+    InherentPermissions = X;
+
+    /// <summary>
+    /// Analyze a single invoice.
+    /// </summary>
+    /// <param name="Base64Data">Data to analyze.</param>
+    /// <param name="CallerModuleInfo">The module info of the caller.</param>
+    /// <returns>The analyzed result.</returns>
+    [NonDebuggable]
+    procedure AnalyzeInvoice(Base64Data: Text; CallerModuleInfo: ModuleInfo) Result: Text
+    var
+        CustomDimensions: Dictionary of [Text, Text];
+    begin
+        AddTelemetryCustomDimensions(CustomDimensions, CallerModuleInfo);
+
+        if not SendRequest(Base64Data, Enum::"ADI Model Type"::Invoice, CallerModuleInfo, Result) then begin
+            FeatureTelemetry.LogError('0000OLK', AzureDocumentIntelligenceCapabilityTok, TelemetryAnalyzeInvoiceFailureLbl, GetLastErrorText(), '', Enum::"AL Telemetry Scope"::All, CustomDimensions);
```

Comment:
Is it possible for GetLastErrorText, which would typically be the error message from ADI, to contain any customer content information?
I don't think so, and it should be fine, but just double check.

Reply:
Might be nice to have the status code of the failure as well.